src/api/Feed.js

```diff
@@ -1128,6 +1261,45 @@ class Feed extends Base {
         }
     };
 
+    /**
+     * Deletes a reply from the cache
+     *
+     * @param {string} id - The id of the feed item to be deleted
+     * @param {string} parentId - The id of the parent feed item
+     * @param {Function} successCallback - function to be called after the delete
+     */
+    deleteReplyItem = (id: string, parentId: string, successCallback: Function = noop) => {
+        const cachedItems = this.getCachedItems(this.file.id);
+
+        const getItemWithFilteredReplies = <T: { replies?: Array<Comment> }>(item: T, replyId: string): T => {
+            if (!item.replies) {
+                return item;
+            }
+            const replies = item.replies.filter(reply => reply.id !== replyId);
+            return { ...item, replies };
+        };
+
+        if (cachedItems) {
+            const feedItems = cachedItems.items.map(item => {
```

Comment:
The separate cache for replies seems a good idea, but I'm not sure if it would be a benefit at this point.

This is my understanding: I think that introducing separate cache for replies would be a complication at this point as the replies are already nested in the initial server response when fetching comments or annotations (last reply is in the replies array) and it's already inserted into the feed items cache. I would need to modify the `feedItems` method and go through all the feed items in order to fill initial value of the `repliesCache`.  In addition I would need to modify the logic of returning the cached feed items by going through all feed items and inserting the replies from the `repliesCache` so that the returned collection can be set as a new value of `feedItems` state (in `ActivitySidebar`). As this would need to be done on every change in the replies (`feedItems` state needs to change to visualize the change in UI) I don't think there will be difference vs the current implementation because mapping the whole collection would need to happen every time state would need to be updated anyway.
I think that having a separate cache for replies would make sense if we would also store the replies as a separate state.

Please let me know if my understanding is incorrect and you had a different solution in mind.

Reply:
I think your understanding is correct. Ideally, the components would be updated to retrieve the correct collection of replies based on the feedItemId passed into them rather than denormalizing the state again. This pattern is very common in [Redux](https://redux.js.org/usage/structuring-reducers/normalizing-state-shape) but can be valuable outside of it, as well.

Since this is the last change we anticipate to the Feed API, we don't need to tackle this now. I do think refactoring this area in the future and normalizing the state would help pull apart some of the spaghetti we're seeing, though.